### PR TITLE
Move math docs out of HelpDB, more examples, fix typos

### PR DESCRIPTION
**Pull request by kshyatt**
Found a series of typos in `cov` and friends. Added more notes
about `NaN` and Julia. Made the function signatures reflect
what's actually in the code. More examples for quite a few
functions.

### CODE REVIEW

File: base/array.jl [hunk 1]
```diff
@@ -658,6 +658,40 @@ hvcat(rows::Tuple{Vararg{Int}}, xs::Union{Vector,Matrix}...) = typed_hvcat(promo
...
+6×3 Array{Float64,2}:
+ 0.397239  0.0369466  0.799969
+ 0.857198  0.507327   0.5634
+ 0.287707  0.367674   0.556441
+ 0.602862  0.835735   0.413742
+ 0.815096  0.36539    0.949436
+ 0.682085  0.179069   0.198193
+```
+"""
 function hcat{T}(V::Vector{T}...)
```
Comment (tkelman) [inline base/array.jl]: this is only for vectors, the docstring is more general

File: base/statistics.jl [hunk 2]
```diff
@@ -493,6 +569,17 @@ end
 median!{T}(v::AbstractArray{T}) = median!(vec(v))
 median{T}(v::AbstractArray{T}) = median!(copy!(Array{T,1}(length(v)), v))
 
+"""
+    median(v[, region])
+
+Compute the median of whole array `v`, or optionally along the dimensions in `region`. For
+even number of elements no exact median element exists, so the result is equivalent to
+calculating mean of two median elements.
```
Comment (tkelman) [inline base/statistics.jl]: some missing articles - "Compute the median of the whole array `v`, or optionally along the dimensions in `region`. For an even number of elements no exact median element exists, so the result is equivalent to calculating the mean of two median elements."

maybe a "the" before the "two" as well?

File: base/rational.jl [hunk 3]
```diff
@@ -90,8 +90,27 @@ promote_rule{T<:Integer,S<:AbstractFloat}(::Type{Rational{T}}, ::Type{S}) = prom
 
 widen{T}(::Type{Rational{T}}) = Rational{widen(T)}
 
+"""
+    rationalize([Type<:Integer,] x; tol::Real=eps(x))
```
Comment (tkelman) [inline base/rational.jl]: I think `T<:Integer` might be more consistent. And should mention that if not provided, `T` defaults to `Int`.

File: base/array.jl [hunk 4]
```diff
@@ -667,6 +701,28 @@ function hcat{T}(V::Vector{T}...)
...
+ 0.574442  0.525314  0.3796    0.486924  0.572558
+
+julia> vcat(a,b)
+3×5 Array{Float64,2}:
+ 0.779721  0.929111  0.721689  0.73127   0.0877032
+ 0.665019  0.692116  0.949552  0.866068  0.0570439
+ 0.574442  0.525314  0.3796    0.486924  0.572558
+```
+"""
 function vcat{T}(arrays::Vector{T}...)
```
Comment (tkelman) [inline base/array.jl]: likewise this signature is only for vectors, docstring is more general

File: base/special/bessel.jl [hunk 5]
```diff
@@ -51,17 +56,53 @@ function airy(k::Integer, z::Complex128)
...
 @vectorize_1arg Number airybiprime
 
+"""
+    airyx(k,x)
+
+scaled `k`th derivative of the Airy function, return ``\\operatorname{Ai}(x) e^{\\frac{2}{3} x \\sqrt{x}}``
+for `k == 0 || k == 1`, and ``\\operatorname{Ai}(x) e^{- \\left| \\operatorname{Re} \\left( \\frac{2}{3} x \\sqrt{x} \\right) \\right|}``
+for `k == 2 || k == 3`.
+"""
 function airyx(k::Integer, z::Complex128)
```
Comment (tkelman) [inline base/special/bessel.jl]: this method is for more specific types than the docstring

File: base/abstractarraymath.jl [hunk 6]
```diff
@@ -11,13 +11,39 @@ transpose(a::AbstractArray) = error("transpose not implemented for $(typeof(a)).
 
 ## Constructors ##
 
+"""
+    vec(a::AbstractArray) -> Vector
+
+Vectorize an array using column-major convention.
```
Comment (tkelman) [inline base/abstractarraymath.jl]: Worth rephrasing this, not immediately obvious what this means.

File: base/math.jl [hunk 7]
```diff
@@ -43,6 +43,21 @@ clamp{X,L,H}(x::X, lo::L, hi::H) =
...
+
+```jldoctest
+julia> clamp([pi, 1.0, big(10.)], 2., 9.)
+3-element Array{BigFloat,1}:
+ 3.141592653589793238462643383279502884197169399375105820974944592307816406286198
+ 2.000000000000000000000000000000000000000000000000000000000000000000000000000000
+ 9.000000000000000000000000000000000000000000000000000000000000000000000000000000
+```
+"""
 clamp{T}(x::AbstractArray{T}, lo, hi) =
```
Comment (tkelman) [inline base/math.jl]: the docstring is more general, there are also methods for scalar input

File: doc/stdlib/arrays.rst [hunk 8]
```diff
@@ -437,12 +437,58 @@ Indexing, Assignment, and Concatenation
 
    Concatenate along dimension 1.
 
+   .. doctest::
+
+       julia> a = rand(1,5)
```
Comment (tkelman) [inline doc/stdlib/arrays.rst]: `rand` isn't a good thing to put in a doctest unless you `srand`

File: base/abstractarraymath.jl [hunk 9]
```diff
@@ -11,13 +11,39 @@ transpose(a::AbstractArray) = error("transpose not implemented for $(typeof(a)).
...
 _sub(::Tuple{}, ::Tuple{}) = ()
 _sub(t::Tuple, ::Tuple{}) = t
 _sub(t::Tuple, s::Tuple) = _sub(tail(t), tail(s))
 
+"""
+    squeeze(A, dims)
+
+Remove the dimensions specified by `dims` from array `A`.
+Elements of `dims` must be unique and within the range `1:ndims(A)`.
+`size(A,i)` must equal 1 for all `i` in `dims.
```
Comment (tkelman) [inline base/abstractarraymath.jl]: missing closing backtick after dims

File: base/statistics.jl [hunk 10]
```diff
@@ -198,12 +220,41 @@ stdm(A::AbstractArray, m::Number; corrected::Bool=true) =
...
 std(iterable; corrected::Bool=true, mean=nothing) =
     sqrt(var(iterable, corrected=corrected, mean=mean))
 
+"""
+    stdm(v, m::Number; corrected::Bool=true)
+
+Compute the sample standard deviation of a vector `v`
+with known mean `m`. If `corrected` is `true`,
+then the sum is scaled with `n-1` whereas the sum is
+scaled with `n` if `corrected` is false` where `n = length(x)`.
```
Comment (tkelman) [inline base/statistics.jl]: missing opening backtick before `false`

**Comment by tkelman** [comment]
Should all the recommendations about DataArrays be changed to recommend NullableArrays now?

**Comment by kshyatt** [comment]
>Should all the recommendations about DataArrays be changed to recommend NullableArrays now?

cc: @johnmyleswhite @davidagold

**Comment by johnmyleswhite** [comment]
Let's change that after DataFrames changes to use NullableArrays. As it stands, we're in a weird intermediate state in which there's no clear standard.

**Pull request merged**